test/parallel/xyk-pallet.API.fees.tripleTokens.test.ts

```diff
@@ -182,180 +134,163 @@ test.skip("xyk-pallet - Check required fee - User with TUR only, operation fails
-    Assets.mintToken(MGA_ASSET_ID, testUser1, new BN(100000)),
+    Assets.mintToken(GASP_ASSET_ID, testUser1, feeLockAmount.divn(2)),
+    Assets.mintToken(ETH_ASSET_ID, testUser1, BN_THOUSAND),
   );
-  let exception = false;
-  await expect(
-    mintLiquidity(
+
+  const saleAssetValue = swapValueThreshold.divn(2);
+
+  try {
+    await signTx(
+      api,
+      Xyk.sellAsset(firstCurrency, secondCurrency, saleAssetValue),
       testUser1.keyRingPair,
-      firstCurrency,
-      secondCurrency,
-      new BN(100),
-      new BN(1000000),
-    ).catch((reason) => {
-      exception = true;
-      throw new Error(reason.data);
-    }),
-  ).rejects.toThrow(feeLockErrors.AccountBalanceFail);
-  expect(exception).toBeTruthy();
+    );
+  } catch (error) {
+    clientError = error;
+  }
+  expect(clientError.data).toContain(
+    "Invalid Transaction: Fee lock processing has failed either due to not enough funds to reserve or an unexpected error",
+  );
 });
 
-test.skip("BUG under discussion:xyk-pallet - when minting all MGX should pay fees with ksm ?- Check required fee - User with very few MGA, some KSM and very few TUR", async () => {
-  await setupApi();
-  await setupUsers();
+test("User, when paying with eth, have to pay 1/10000000 eth per GASP spent.", async () => {
+  const [testUser2] = setupUsers();
+
+  testUser2.addAsset(GASP_ASSET_ID);
+  testUser2.addAsset(ETH_ASSET_ID);
+
   await Sudo.batchAsSudoFinalized(
-    Assets.mintToken(TUR_ASSET_ID, testUser1, new BN(100000)),
-    Assets.mintToken(KSM_ASSET_ID, testUser1),
-    Assets.mintToken(MGA_ASSET_ID, testUser1, Assets.DEFAULT_AMOUNT),
-    Assets.mintToken(secondCurrency, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(firstCurrency, testUser2, defaultCurrencyValue),
+    Assets.mintToken(secondCurrency, testUser2, defaultCurrencyValue),
+    Assets.mintToken(ETH_ASSET_ID, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser1, Assets.DEFAULT_AMOUNT),
   );
-  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await runMintingLiquidity(testUser1);
+  await runMintingLiquidity(testUser2);
+
+  const deductedGaspTkns = await getDeductedTokens(testUser1, GASP_ASSET_ID);
+  const deductedEthTkns = await getDeductedTokens(testUser2, ETH_ASSET_ID);
+  const feesRatio = deductedGaspTkns.div(deductedEthTkns).toNumber();
+
+  expect(feesRatio).toEqual(10000000);
+  expect(deductedGaspTkns).bnGt(BN_ZERO);
+  expect(deductedEthTkns).bnGt(BN_ZERO);
+});
+
+async function getDeductedTokens(testUser: User, tokenId: BN) {
+  const deductedTokens = testUser
+    .getAsset(tokenId)!
+    .amountBefore.free.sub(testUser.getAsset(tokenId)!.amountAfter.free!);
+  return deductedTokens;
+}
+
+async function runMintingLiquidity(testUser: User) {
+  await testUser.refreshAmounts(AssetWallet.BEFORE);
 
   await mintLiquidity(
-    testUser1.keyRingPair,
-    MGA_ASSET_ID,
+    testUser.keyRingPair,
+    firstCurrency,
```

Comment:
send this as a param, NEVER from a global variable.

Reply:
Fixed